pinot-core/src/main/java/com/linkedin/pinot/core/common/DataFetcher.java

```diff
@@ -107,6 +119,42 @@ public void fetchSingleDictIds(String column, int[] inDocIds, int inStartPos, in
   }
 
   /**
+   * Fetch the dictionary Ids for a multi value column.
+   *
+   * @param column column name.
+   * @param inDocIds document Id array.
+   * @param inStartPos input start position.
+   * @param length input length.
+   * @param outDictIdsArray dictionary Id array array buffer.
+   * @param outStartPos output start position.
+   */
+  public void fetchMultiValueDictIds(String column, int[] inDocIds, int inStartPos, int length, int[][] outDictIdsArray,
+      int outStartPos) {
+    BlockMultiValIterator iterator = (BlockMultiValIterator) getBlockValSetForColumn(column).iterator();
+    int maxNumberOfEntries = getMaxNumberOfEntriesForColumn(column);
+    int[] dictIdArray = new int[maxNumberOfEntries];
```

Comment:
This is per block one. Then I will change fetchMultiValueDictIds to allow passing in an int[] for dictIdArray